aiida_testing/mock_code/_fixtures.py

```diff
@@ -8,33 +8,91 @@
 import inspect
 import pathlib
 import typing as ty
-
+import click
 import pytest
 
+from aiida.orm import Code
+
 from ._env_keys import EnvKeys
-from .._config import get_config
+from .._config import Config, CONFIG_FILE_NAME, ConfigActions
+
+__all__ = (
+    "pytest_addoption",
+    "testing_config_action",
+    "mock_regenerate_test_data",
+    "testing_config",
+    "mock_code_factory",
+)
+
+
+def pytest_addoption(parser):
+    """Add pytest command line options."""
+    parser.addoption(
+        "--testing-config-action",
+        type=click.Choice((c.value for c in ConfigActions)),
+        default=ConfigActions.READ.value,
+        help=f"Read {CONFIG_FILE_NAME} config file if present ('read'), require config file ('require') or " \
+             "generate new config file ('generate').",
+    )
+    parser.addoption(
+        "--regenerate-test-data", action="store_true", default=False, help="Regenerate test data."
+    )
+
+
+@pytest.fixture(scope='session')
+def testing_config_action(request):
+    return request.config.getoption("--testing-config-action")
+
+
+@pytest.fixture(scope='session')
+def mock_regenerate_test_data(request):
+    return request.config.getoption("--regenerate-test-data")
+
+
+@pytest.fixture(scope='session')
+def testing_config(testing_config_action):  # pylint: disable=redefined-outer-name
+    """Get content of .aiida-testing-config.yml
+
+    testing_config_action :
+        Read config file if present ('read'), require config file ('require') or generate new config file ('generate').
+    """
+    config = Config.from_file()
 
-__all__ = ("mock_code_factory", )
+    if not config and testing_config_action == ConfigActions.REQUIRE.value:
+        raise ValueError(f"Unable to find {CONFIG_FILE_NAME}.")
+
+    yield config
+
+    if testing_config_action == ConfigActions.GENERATE.value:
+        config.to_file()
 
 
 @pytest.fixture(scope='function')
-def mock_code_factory(aiida_localhost):
+def mock_code_factory(
+    aiida_localhost, testing_config, testing_config_action, mock_regenerate_test_data
+):  # pylint: disable=redefined-outer-name
     """
     Fixture to create a mock AiiDA Code.
-    """
-    config = get_config().get('mock_code', {})
 
+    testing_config_action :
+        Read config file if present ('read'), require config file ('require') or generate new config file ('generate').
+
+
+    """
     def _get_mock_code(
         label: str,
         entry_point: str,
         data_dir_abspath: ty.Union[str, pathlib.Path],
-        ignore_files: ty.Iterable[str] = ('_aiidasubmit.sh', )
-    ):
+        ignore_files: ty.Iterable[str] = ('_aiidasubmit.sh'),
+        executable_name: str = '',
+        config: dict = testing_config,
```

Comment:
Are `config`, `config_action` and `regenerate_test_data` here passed just for being able to test the fixture internally, or do we expect users to set them?

I'm not really sure if there's an established convention for that, but maybe we could indicate this by a leading underscore in the kwarg names (`_config`, `_config_action`, `_regenerate_test_data`)?

Reply:
done!